contracts/strategies/stkCvxFXS/StrategyBase.sol

```diff
@@ -0,0 +1,463 @@
+            .ExactInputParams(
+                abi.encodePacked(WETH_TOKEN, fee, USDC_TOKEN, fee, FRAX_TOKEN),
+                address(this),
+                block.timestamp + 1,
+                _amount,
+                0
+            );
+
+        uint256 _fraxAmount = IUniV3Router(UNIV3_ROUTER).exactInput{
+            value: _amount
+        }(_params);
+        address[] memory _path = new address[](2);
+        _path[0] = FRAX_TOKEN;
+        _path[1] = FXS_TOKEN;
+        uint256[] memory amounts = IUniV2Router(UNISWAP_ROUTER)
+            .swapExactTokensForTokens(
+                _fraxAmount,
+                1,
+                _path,
+                address(this),
+                block.timestamp + 1
+            );
+        return amounts[1];
+    }
+
+    /// @notice Swap FXS->ETH on UniV3 via stable pair
+    /// @param _amount - amount to swap
+    /// @return amount of token obtained after the swap
+    function _uniStableFxsToEthSwap(uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        address[] memory _path = new address[](2);
+        _path[0] = FXS_TOKEN;
+        _path[1] = FRAX_TOKEN;
+        uint256[] memory amounts = IUniV2Router(UNISWAP_ROUTER)
+            .swapExactTokensForTokens(
+                _amount,
+                1,
+                _path,
+                address(this),
+                block.timestamp + 1
+            );
+
+        uint256 _fraxAmount = amounts[1];
+        uint24 fee = 500;
+
+        IUniV3Router.ExactInputParams memory _params = IUniV3Router
+            .ExactInputParams(
+                abi.encodePacked(FRAX_TOKEN, fee, USDC_TOKEN, fee, WETH_TOKEN),
+                address(this),
+                block.timestamp + 1,
+                _fraxAmount,
+                0
+            );
+
+        uint256 _ethAmount = IUniV3Router(UNIV3_ROUTER).exactInput{value: 0}(
+            _params
+        );
+        IWETH(WETH_TOKEN).withdraw(_ethAmount);
+        return _ethAmount;
+    }
+
+    /// @notice Swap FXS->ETH on a mix of UniV2, UniV3 & Curve
+    /// @param _amount - amount to swap
+    /// @return amount of token obtained after the swap
+    function _uniCurve1FxsToEthSwap(uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        address[] memory _path = new address[](2);
+        _path[0] = FXS_TOKEN;
+        _path[1] = FRAX_TOKEN;
+        uint256[] memory amounts = IUniV2Router(UNISWAP_ROUTER)
+            .swapExactTokensForTokens(
+                _amount,
+                1,
+                _path,
+                address(this),
+                block.timestamp + 1
+            );
+
+        uint256 _fraxAmount = amounts[1];
+        // Swap FRAX for USDC on Curve
+        uint256 _usdcAmount = fraxUsdcSwap.exchange(0, 1, _fraxAmount, 0);
+
+        // USDC to ETH on UniV3
+        uint24 fee = 500;
+        IUniV3Router.ExactInputParams memory _params = IUniV3Router
```

Comment:
It may be worth creating a pure helper method for producing this struct (instances throughout the code and a few of the fields appear to be fixed) to dedupe code, help readability, and prevent potential errors in the future.